### PR TITLE
[CI] Allow to pass the xa checkout dir to nested templates.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: build-tools/automation/yaml-templates/run-msbuild-tests.yaml [hunk 1]
```diff
@@ -31,7 +32,8 @@ jobs:
       installLegacyDotNet: false
       restoreNUnitConsole: false
       updateMono: false
-      
+      xaSourcePath: ${{ parameters.xaSourcePath }}
```
Comment (pjcollins) [inline build-tools/automation/yaml-templates/run-msbuild-tests.yaml:35]: Do we also need to change https://github.com/xamarin/xamarin-android/blob/09daa0cf0f2f5dc366f04b4bdc61c6a758cb59d6/build-tools/automation/yaml-templates/setup-test-environment.yaml#L15 to use the checkout template?

Comment (mandel-macaque) [reply build-tools/automation/yaml-templates/run-msbuild-tests.yaml:35]: Correct, that will need to be done too, I did not reach that point yet, we probably want a way to use the template and pass it along like in all others

File: build-tools/automation/yaml-templates/setup-test-environment.yaml [hunk 2]
```diff
@@ -10,13 +10,19 @@ parameters:
   restoreNUnitConsole: true
   updateMono: true
   androidSdkPlatforms: $(DefaultTestSdkPlatforms)
+  repositoryAlias: 'self'
+  commit: 'HEAD'
```
Comment (pjcollins) [inline build-tools/automation/yaml-templates/setup-test-environment.yaml:14]: Can we use an empty string here by default instead, like we're doing in the [build templates](https://github.com/xamarin/xamarin-android/blob/main/build-tools/automation/yaml-templates/build-macos.yaml#L5)? Same thing applies to the other templates

Comment (mandel-macaque) [inline build-tools/automation/yaml-templates/setup-test-environment.yaml:14]:
```suggestion
  commit: ''
```

File: build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml [hunk 3]
```diff
@@ -7,6 +7,9 @@ parameters:
   agent_count: 8
   stageCondition: succeeded()
   stagePrefix: ''
+  xaSourcePath: $(System.DefaultWorkingDirectory)
+  repositoryAlias: 'self'
+  commit: 'HEAD'
```
Comment (mandel-macaque) [inline build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml:12]:
```suggestion
  commit: ''
```

**Review by pjcollins — APPROVED** [review]

**Pull request merged**